docs/content/direct/wec-registration.md

```diff
@@ -1,5 +1,307 @@
 # Registering a Workload Execution Cluster
 
-This document will tell users what how to register a WEC in an ITS.
+This document explains how to register a Workload Execution Cluster (WEC) with an Inventory and Transport Space (ITS) in KubeStellar.
 
-TODO: write this.
+## Overview
+
+Registering a WEC in an ITS is the same process as registering a managed cluster with an OCM hub cluster. KubeStellar uses Open Cluster Management (OCM) for cluster registration and management.
+
+
+### Terminology Mapping
+
+| OCM Term             | KubeStellar Equivalent                        |
+|----------------------|-----------------------------------------------|
+| **OCM Hub**          | **KubeStellar ITS** (Inventory and Transport Space) |
+| **OCM Managed Cluster** | **KubeStellar WEC** (Workload Execution Cluster) |
+| **OCM Agent**        | **OCM Agent** (same component, installed on WEC) |
+
+For the complete OCM registration process, refer to the [official Open Cluster Management documentation](https://open-cluster-management.io/docs/getting-started/installation/register-a-cluster/).
+
+## Prerequisites
+
+Before registering a WEC, ensure you have:
+
+1. **A running Kubernetes cluster** that will serve as the WEC
+2. **Network connectivity** between the WEC and the ITS
+3. **`kubectl` access** to both the WEC and ITS
+4. **`clusteradm` CLI tool** installed ([installation guide](https://open-cluster-management.io/docs/getting-started/installation/start-the-control-plane/))
+5. **An existing ITS** with OCM cluster manager running
+6. **Sufficient permissions** to create resources in both clusters
+
+To verify your ITS is ready for WEC registration:
+
+```shell
+# Check if the ITS is accessible (this returns a short result)
+kubectl --context <its-context> get ServiceAccount default
+
+# Verify OCM cluster manager is running (look for pods in Running state)
+kubectl --context <its-context> get pods -n open-cluster-management-hub
+
+# Check for the customization-properties namespace
+kubectl --context <its-context> get ns customization-properties
+```
+
+## Registration Process
+
+### Step 1: Get the Registration Token
+
+Obtain a registration token from the ITS:
+
+```shell
+# Get the registration token from the ITS
+clusteradm --context <its-context> get token
+```
+
+This command outputs a `clusteradm join` command that you'll use in the next step.
+
+### Step 2: Join the WEC to the ITS
+
+Execute the join command on the WEC to initiate the registration process:
+
+```shell
+# Basic join command
+clusteradm join --hub-token <token> --hub-apiserver <api-server-url> --cluster-name <your-wec-name> --context <wec-context>
+```
```

Comment:
This does not hint that there might be additional flags required; that could lead an eager reader to make a mistake.

Reply:
The current revision still starts by telling the reader to do something without warning about `--force-internal-endpoint-lookup`.